daemon/daemon.go

```diff
@@ -310,6 +310,8 @@ func (daemon *Daemon) restore() error {
 				}
 			}
 
+			// reset container restart count
+			c.RestartManager(true)
```

Comment:
RestartManager decides whether the container should be restarted based on its `failureCount`,  If we don't reset container's restart count, a container with `on-failure:3` may have a `RestartCount` 5. e.g.
1. container failed 2 times, `RestartCount`=2
2. daemon shutdown.
3. daemon reboot, *RestartCount not reset*, container failed and restart 3 times, RestartCount turns out to be 2+3=5.  

This is incorrect.